### PR TITLE
SPARK-12868: Allow Add jar to add jars from hdfs/s3n urls.

### PR DESCRIPTION
**Pull request by Parth-Brahmbhatt**
## What changes were proposed in this pull request?

Add jar command fails when given s3n or hdfs urls. This changes fixes that.
## How was this patch tested?

unit test added, it was just copied from a previously opened PR https://github.com/apache/spark/pull/10797 so credit should be given to the person who originally posted the patch.

### CODE REVIEW

File: sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala
Comment (jaceklaskowski) [inline sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala]: Why do you need `{}`?

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveQuerySuite.scala
Comment (jaceklaskowski) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveQuerySuite.scala]: You sure it's `nn1` (with `1`)?

File: sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala
Comment (Parth-Brahmbhatt) [inline sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala]: Removed.

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveQuerySuite.scala
Comment (Parth-Brahmbhatt) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveQuerySuite.scala]: Good catch, unintentional typo. fixed.

**Comment by Parth-Brahmbhatt** [comment]
Can one of the committers take a look at this PR?

**Comment by Parth-Brahmbhatt** [comment]
Can one of the committers take a look at this PR?

**Comment by Parth-Brahmbhatt** [comment]
Request for review one more time.

**Comment by Parth-Brahmbhatt** [comment]
Ping.

**Comment by Parth-Brahmbhatt** [comment]
Request for review.

**Comment by felixcheung** [comment]
@srowen would this make sense to you?

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveQuerySuite.scala
Comment (rdblue) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveQuerySuite.scala]: I think this test should be improved before merging this. Looking for a RuntimeException to validate that the Jar was registered is brittle and can easily pass when the registration doesn't actually work.

**Review by rdblue — CHANGES_REQUESTED** [review]

**Comment by HyukjinKwon** [comment]
Hi @Parth-Brahmbhatt, is it still active? If so is there any argument against https://github.com/apache/spark/pull/14720#discussion_r87908473?

**Comment by wangyum** [comment]
It has been fixed by https://github.com/apache/spark/pull/17342.

**Comment by HyukjinKwon** [comment]
I will propose to close this after a week if it is still inactive or no objection to ^.

**Comment by Parth-Brahmbhatt** [comment]
closing.

**Pull request closed without merging**
